test/tools/d_do_test.d

```diff
@@ -1012,23 +1071,16 @@ int tryMain(string[] args)
     if (envData.coverage_build && testArgs.mode == TestMode.RUN)
         testArgs.mode = TestMode.COMPILE;
 
+    // FIXME: Handle this in processEnvironment
     if (envData.ccompiler.empty)
     {
-        if (envData.os != "windows")
-            envData.ccompiler = "c++";
-        else if (envData.model == "32")
-            envData.ccompiler = "dmc";
-        else if (envData.model == "64")
-            envData.ccompiler = `C:\"Program Files (x86)"\"Microsoft Visual Studio 10.0"\VC\bin\amd64\cl.exe`;
-        else
-            assert(0, "unknown $OS$MODEL combination: " ~ envData.os ~ envData.model);
+        writeln("Unknown $OS$MODEL combination: ", envData.os, envData.model);
+        throw new SilentQuit();
```

Comment:
Eagerly aborting in `processEnvironment` somehow fails on many CI machines, hence I'll keep this for now

Reply:
Which ones ? Is it consistent ?

Reply:
Most of them. And yes.

Reply:
Really? Its all green, is this good to go then?

Reply:
I don't recall which ones exactly but I would guess it concerns platforms which don't usually run the C++ tests